src/Polly.Core/Simmy/Behavior/BehaviorCompositeStrategyBuilderExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Polly.Simmy.Behavior;
+
+/// <summary>
+/// Extension methods for adding custom behaviors to a <see cref="CompositeStrategyBuilder"/>.
+/// </summary>
+public static class BehaviorCompositeStrategyBuilderExtensions
+
+{
+    /// <summary>
+    /// Adds a latency chaos strategy to the builder.
+    /// </summary>
+    /// <typeparam name="TBuilder">The builder type.</typeparam>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="enabled">A value that indicates whether or not the chaos strategy is enabled for a given execution.</param>
+    /// <param name="injectionRate">The injection rate for a given execution, which the value should be between [0, 1].</param>
+    /// <param name="behavior">The behavior to be injected.</param>
+    /// <returns>The same builder instance.</returns>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="builder"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ValidationException">Thrown when the options produced from the arguments are invalid.</exception>
+    public static TBuilder AddBehavior<TBuilder>(this TBuilder builder, bool enabled, double injectionRate, Func<ValueTask> behavior)
```

Comment:
Prefix with `Chaos` i.e. `AddChaosBehavior` to make it clear this is in a family of chaos strategies?

(applies across all extensions)